fix '@theia' implicit dependencies

vince-fugnitto opened the pull request:
<!--
Thank you for your Pull Request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/theia-ide/theia/blob/master/CONTRIBUTING.md
-->

#### What it does
<!-- Include relevant issues and describe how they are addressed -->

Fixes #5790

- fixed the `@theia/` `no-implicit-dependencies` warnings from extensions
which used dependencies directly without declaring them in their package.json
- fixed issue in `@theia/plugin-ext` regarding incorrect import paths

#### How to test
1. build an application that uses `@theia/plugin-ext` or `@theia/plugin-ext-vscode` excluding `@theia/mini-browser` as a dependency. (ensure that `@theia/mini-browser` is not being pulled by another dependency as well.

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully reviewed following [the review guidelines](https://github.com/theia-ide/theia/blob/ak/pr_template/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/ak/pr_template/doc/pull-requests.md#reviewing)

Signed-off-by: Vincent Fugnitto <vincent.fugnitto@ericsson.com>

<!-- Please provide a clear and meaningful description to the CHANGELOG.md file if this PR contributes some significant changes -->

vince-fugnitto commented:
@marcdumais-work what's unfortunate is that `mini-browser` is only being pulled for a css class name.

https://github.com/theia-ide/theia/blob/22ef59c5415ec45bacac6bd5ddab5ef675d35cf7/packages/plugin-ext/src/main/browser/webview/webview.ts#L58

marcdumais-work commented:
> unfortunate is that mini-browser is only being pulled for a css class name.

True. Removing the dependency would also be a valid fix, but I'll let @benoitf and @evidolob decide what they think is best in this case.

akosyakov commented:
We will need to move class somewhere else then, let's add it for now. It cannot work otherwise.

akosyakov commented on packages/plugin-ext/src/plugin/languages/rename.ts:
```diff
@@ -18,9 +18,8 @@ import URI from 'vscode-uri/lib/umd';
 import * as theia from '@theia/plugin';
 import * as Converter from '../type-converters';
 import * as model from '../../api/model';
-import { DocumentsExtImpl } from '@theia/plugin-ext/src/plugin/documents';
```
that's a good catch, how did you find it?

vince-fugnitto replied:
tslint was complaining that `@theia/plugin-ext` was implicitly depending on itself so I 
checked as to how/why :see_no_evil:

akosyakov replied:
Was it a warning as well, it would be good to have it as error.

vince-fugnitto replied:
It was a warning, I guess we need a custom rule to find out if an extension is trying to import istelf this way.

vince-fugnitto commented:
@benoitf are you fine with me merging?

akosyakov commented:
@vince-fugnitto @marcdumais-work We don't have code ownership. Everybody should be able to change, review and discuss whatever comes. We've added @evidolob @benoitf because they wanted to be notified about changes in the plugin system. It does not mean that you need an approve. One approve from anybody who verified design and tested changes is fine to merge. You also don't need approve from me on PRs to core. I will comment if i see something suspicious.

Let's remove code owners file if it is confusing. I see already several PRs where it happens and no response. We could have a rule instead that any nontrivial PR should sit at least 2 days after being approved. So everybody for sure had time to have a look. Also if it breaks we revert it within 2 days.

Should we discuss it on next dev meeting?

vince-fugnitto commented:
> We've added @evidolob @benoitf because they wanted to be notified about changes in the plugin system. 

Sounds good, inititally I thought it meant that were required to review changes from the plugin system (based on past PRs) but if its meant to only notify them of changes then I'll keep that in mind for the future :)

> You also don't need approve from me on PRs to core.

Yes I understand. Generally I feel more comfortable adding you to reviews that are significant or may require more in depth design decisions since you have valuable insight. I'd rather wait for a PR to be complete and implemented well before I merge.

> Let's remove code owners file if it is confusing. I see already several PRs where it happens and no response. We could have a rule instead that any nontrivial PR should sit at least 2 days after being approved. So everybody for sure had time to have a look. Also if it breaks we revert it within 2 days.

I think it's fine, I know they wanted to be notified of changes to the plugin system but with the newer ways of working I understand that no one has code ownership anymore.

> Should we discuss it on next dev meeting?

If you'd like to bring it up I'd be fine with that.
